src/main/java/core/basesyntax/CsvFileWriter.java

```
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class CsvFileWriter extends FileDataWriter {
+    @Override
+    public void write(String filePath, String data) {
+        if (filePath == null || filePath.isEmpty()) {
+            throw new RuntimeException("Failed to write data to a file. Path is empty");
+        }
+
+        Path path = Paths.get(filePath);
+        Path parentDir = path.getParent();
+
+        if (parentDir != null && !Files.exists(parentDir)) {
+            try {
+                Files.createDirectories(parentDir);
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to create directory " + parentDir);
+            }
+        }
+
+        try {
+            Files.writeString(path, data, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to write the file " + path);
```

This violates checklist item #6: 'Don't ignore exceptions in try-catch blocks'. To preserve the original error information, you should pass the caught exception `e` as the second argument to the `RuntimeException` constructor, like this: `throw new RuntimeException("Failed to write the file " + path, e);`.